incubator/hnc/internal/reconcilers/object.go

```diff
@@ -763,7 +764,13 @@ func (r *ObjectReconciler) syncUnpropagatedSource(ctx context.Context, log logr.
 // - Objects of a type whose mode is set to "remove" in the HNCConfiguration singleton
 // - Objects with nonempty finalizer list
 // - Service Account token secrets
-func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstructured.Unstructured) bool {
+func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst, destInst *unstructured.Unstructured) bool {
+	var selector labels.Selector
+	var nsLabels labels.Set
+	if destInst != nil {
+		selector = r.getSelector(log, inst)
+		nsLabels = labels.Set(r.Forest.Get(destInst.GetNamespace()).GetLabels())
```

Comment:
Should `getLabels()` return a `labels.Set`, if that's the only way this is ever used?

Reply:
done